src/quo/components/list_items/approval_info/view.cljs

```diff
@@ -0,0 +1,118 @@
+(ns quo.components.list-items.approval-info.view
+  (:require [clojure.string :as string]
+            [quo.components.avatars.account-avatar.view :as account-avatar]
+            [quo.components.avatars.collection-avatar.view :as collection-avatar]
+            [quo.components.avatars.community-avatar.view :as community-avatar]
+            [quo.components.avatars.dapp-avatar.view :as dapp-avatar]
+            [quo.components.avatars.icon-avatar :as icon-avatar]
+            [quo.components.avatars.token-avatar.view :as token-avatar]
+            [quo.components.avatars.wallet-user-avatar.view :as wallet-user-avatar]
+            [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icon]
+            [quo.components.list-items.approval-info.style :as style]
+            [quo.components.markdown.text :as text]
+            [quo.components.tags.tiny-tag.view :as tiny-tag]
+            [quo.foundations.colors :as colors]
+            quo.theme
+            [react-native.core :as rn]
+            [schema.core :as schema]))
+
+(def ?schema
+  [:=>
+   [:catn
+    [:props
+     [:map {:closed true}
+      [:type
+       [:enum :spending-cap :token-contract :account :spending-contract :network :date-signed
+        :collectible :address :community]]
+      [:avatar-props :map]
+      [:label :string]
+      [:description {:optional true} [:maybe :string]]
+      [:button-label {:optional true} [:maybe :string]]
+      [:button-icon {:optional true} [:maybe :keyword]]
+      [:blur? {:optional true} [:maybe :boolean]]
+      [:unlimited-icon? {:optional true} [:maybe :boolean]]
+      [:option-icon {:optional true} [:maybe :keyword]]
+      [:tag-label {:optional true} [:maybe :string]]
+      [:on-button-press {:optional true} [:maybe fn?]]
+      [:on-avatar-press {:optional true} [:maybe fn?]]
+      [:on-option-press {:optional true} [:maybe fn?]]
+      [:container-style {:optional true} [:maybe :any]]]]]
+   :any])
+
+(defn- avatar
+  [{:keys [type avatar-props blur? theme on-press]}]
+  [rn/pressable {:on-press on-press}
+   (cond
+     (#{:spending-cap :token-contract :network
+        :collectible}
```

Comment:
I think case will be better choice here. wdyt?

Reply:
It is way better with `case`, thanks for the suggestions 😊